Libraries/PeriphDrivers/Source/SKBD/skbd_reva.c

```diff
@@ -119,11 +119,10 @@ int MXC_SKBD_RevA_Init(mxc_skbd_reva_regs_t *skbd, mxc_skbd_config_t config)
         }
 
         /* Configure the SKBD  */
-        temp =
-            ((config.reg_erase << MXC_F_SKBD_REVA_CTRL1_CLEAR_POS) | MXC_F_SKBD_REVA_CTRL1_AUTOEN);
-        temp |= (config.debounce << MXC_F_SKBD_REVA_CTRL1_DBTM_POS) & MXC_F_SKBD_REVA_CTRL1_DBTM;
-        temp |= (outputs << MXC_F_SKBD_REVA_CTRL1_OUTNB_POS) & MXC_F_SKBD_REVA_CTRL1_OUTNB;
-        skbd->ctrl1 |= temp;
+        skbd->ctrl1 = MXC_F_SKBD_REVA_CTRL1_AUTOEN // ->ctrl1[0]
+                		| MXC_F_SKBD_REVA_CTRL1_CLEAR // ->ctrl1[1]
+                		| MXC_F_SKBD_REVA_CTRL1_OUTNB //ctrl1[11:8]
+                		| config.debounce << MXC_F_SKBD_REVA_CTRL1_DBTM_POS; //ctrl1[15:13]
```

Comment:
`config.debounce` is set as a 32-bit struct member, but the `SKBD_CTRL1.dbtm` field is 3-bits wide. It shouldn't be breaking if the user passes in a value greater than 3 bits because bits [31:16] are reserved in the `SKBD_CTRL1` register. However, it'd be good practice to mask this in case design adds more fields to this register even if it's unlikely.

Reply:
Fixed.